Python/Product/Cookiecutter/Model/CookiecutterClient.cs

```diff
@@ -148,27 +148,30 @@ class CookiecutterClient : ICookiecutterClient {
             // Example:
             // "_visual_studio" : {
             //   "var1" : {
-            //     "description" : "description for var 1",
+            //     "label" : "Variable 1",
+            //     "description" : "Description for variable 1",
             //     "selector" : "yesno"
             //   },
             //   "var2" : {
-            //     "description" : "description for var 2",
-            //     "selector" : "connection"
+            //     "label" : "Variable 2",
+            //     "description" : "Description for variable 2",
+            //     "selector" : "odbcConnection"
             //   }
             // }
             //
             // Valid values for selector:
             // - string
             // - list
             // - yesno: generates 'y' or 'n'
-            // - connection
+            // - odbcConnection
             if (vsExtrasProp.Value.Type == JTokenType.Object) {
                 var vsExtrasObj = (JObject)vsExtrasProp.Value;
                 foreach (JProperty prop in vsExtrasObj.Properties()) {
                     var item = items.SingleOrDefault(ctx => ctx.Name == prop.Name);
                     if (item != null) {
                         if (prop.Value.Type == JTokenType.Object) {
                             var itemObj = (JObject)prop.Value;
+                            ReadLabel(item, itemObj);
```

Comment:
ContextItemViewModel ctor sets it to label if not null/empty, otherwise param name.